flatdata-rs/lib/src/vector.rs

```diff
@@ -53,65 +49,45 @@ use std::{borrow::BorrowMut, fmt, io, marker};
 ///
 /// let archive = X::open(storage).expect("failed to open");
 /// let view = archive.data();
-/// assert_eq!(view.at(1).x(), 3);
+/// assert_eq!(view[1].x(), 3);
 /// ```
 ///
 /// [`ExternalVector`]: struct.ExternalVector.html
 /// [`as_view`]: #method.as_view
-#[derive(Clone)]
 pub struct Vector<T>
 where
-    T: RefFactory,
+    T: Struct,
 {
-    data: Vec<u8>,
-    _phantom: marker::PhantomData<T>,
+    data: Vec<T>,
 }
 
 impl<T> Vector<T>
 where
-    T: RefFactory,
+    T: Struct,
 {
-    fn padding() -> usize {
-        if <T as Struct>::IS_OVERLAPPING_WITH_NEXT {
-            <T as Struct>::SIZE_IN_BYTES + memory::PADDING_SIZE
-        } else {
-            memory::PADDING_SIZE
-        }
-    }
     /// Creates an empty `Vector<T>`.
     #[inline]
     pub fn new() -> Self {
         Self::with_len(0)
     }
 
+    /// Resets the 'Vector<T>' to its initial empty state
+    #[inline]
+    pub fn clear(&mut self) {
+        self.data.clear();
+        self.data.push(unsafe { T::create_unchecked() });
+    }
+
     /// Creates a `Vector<T>` with `len` many elements.
     ///
     /// `T`'s fields are all filled with zeroes.
     #[inline]
     pub fn with_len(len: usize) -> Self {
-        let size = Self::calc_size(len);
-        Self {
-            data: vec![0; size],
-            _phantom: marker::PhantomData,
+        let mut data = Vec::with_capacity(len + 1);
+        for _ in 0..=len {
+            data.push(unsafe { T::create_unchecked() });
         }
-    }
-
-    /// Size of the vector in bytes.
-    #[inline]
-    pub fn size_in_bytes(&self) -> usize {
-        self.data.len() - Self::padding()
-    }
-
-    /// Number of elements in the vector.
-    #[inline]
-    pub fn len(&self) -> usize {
```

Comment:
Any reason for removing this?

Reply:
You get it for free from Deref<&[T]> (like most other slice functions I removed)